### ISSUE TITLE
CSet creation

### ISSUE THREAD

**Issue by olynch** [AlgebraicJulia/Catlab.jl#205]
It would be interesting to define a CSet using "copattern matching". That is, by defining it with syntax that harkens to the fact that it is a functor from the indexing category to Set. For instance, to define a graph, one could have

```julia
g(:V) = FinOrd(2)
g(:E) = FinOrd(1)
g(:src) = FinOrdFunction([1],2)
g(:tgt) = FinOrdFunction([2],2)
```

wrapped in a suitable macro. This is not high priority, I'm just writing this down so I can refer back to it.

**Comment by jpfairbanks** [comment]
That is a great idea. Would this be a good candidate for the macro syntax?

```julia
g = @cset TheoryGraph => (FinOrd, FinOrdFunction)
  V = 2
  E = 1
  src = ([1], 2)
  tgt = ([2], 2)
end
```

It relies on the macro to know that src,tgt are homs and so should get mapped to `FinOrdFunction([1],2)`, `FinOrdFunction([1],2)`.

**Comment by olynch** [comment]
I'm going to take a whack at this, because this would be a nice thing to write up in the paper.

**Comment by epatters** [comment]
Resolved by #333.
